feat: 로그인 페이지에 카카오 로그인 api 로직 추가

smb0123 opened the pull request:
### 개요

close #6 

### 작업사항

- tanstack query, devtools 설치 및 초기 세팅
- axios 설치 후 공통 api 함수 구현 (https)
- 카카오 로그인 api 로직을 추가하였습니다.

mutate 로직을 가진 useKakaoLoginMutate를 만들고 postKakaoLogin함수를 리턴 한 후 useKakaoLogin에서 사용하도록 코드를 작성하였습니다. 이는 아래와 같은 장점이 있습니다.
1. useKakaoLoginMutate 훅을 재사용이 가능해집니다.
2. 함수형으로 작성하여 가독성이 좋습니다.

- 소셜로그인이 성공하면 토큰을 저장할 social-login-loading 페이지 구현

useSaveToken 훅에서는 쿼리 파라미터를 추출하여 로컬 스토리지에 저장하고 저장이 완료되면 특정 페이지로 이동시켰습니다.

### 변경로직

- 애플 로그인 버튼, 카카오 로그인 버튼 컴포넌트를 구현하였습니다.


### 체크리스트

- [x] assignees 설정
- [x] label 설정

seunghyeon77 committed:
chore: clsx, cva, tailwind merge 라이브러리 설치

seunghyeon77 committed:
feat: 이메일 인증 페이지 추가

seunghyeon77 committed:
feat: 이메일 인증 Section 구현

seunghyeon77 committed:
feat: Text ui 컴포넌트 추가

seunghyeon77 committed:
feat: Input ui 컴포넌트 추가

seunghyeon77 committed:
feat: Button ui 컴포넌트 추가

seunghyeon77 committed:
feat: Spacing ui 컴포넌트 추가

seunghyeon77 committed:
feat: 전역 색상 저장 및 동적으로 불러올 값 저장

seunghyeon77 committed:
design: 이메일 인증 Section이 아닌 이메일 인증 페이지에서 페이지 전체 margin 되도록 수정

seunghyeon77 committed:
chore: package-lock.json 파일 추가

seunghyeon77 committed:
feat: Spacing, Button UI 컴포넌트로 변경

seunghyeon77 committed:
refactor: div태그에서 Fragment 태그로 변경

seunghyeon77 committed:
chore: yellow 이름을 kakao-yellow로 구체화

seunghyeon77 committed:
refactor: Flex layout 컴포넌트 삭제

smb0123 committed:
chore: section 태그 추가

컴포넌트명에 맞게 section 태그를 추가하였습니다.

smb0123 committed:
chore: section 태그 추가

컴포넌트명에 맞게 section 태그를 추가하였습니다.

smb0123 committed:
refactor: 애플 로그인 버튼, 카카오 로그인 버튼 컴포넌트로 분리

가독성과 로직의 분리를 위해서 변경하였습니다.

smb0123 committed:
refactor: 애플 로그인 버튼, 카카오 로그인 버튼 컴포넌트로 분리

가독성과 로직의 분리를 위해서 변경하였습니다.

smb0123 committed:
chore: axios 라이브러리 설치

api 통신을 원활하게 하기 위해서 설치하였습니다.

smb0123 committed:
chore: axios 라이브러리 설치

api 통신을 원활하게 하기 위해서 설치하였습니다.

smb0123 committed:
chore: tanstack query와 devtools 설치

api 통신 로직을 간결하게 하고 디버깅하기 위해서 설치하였습니다.

smb0123 committed:
chore: tanstack query와 devtools 설치

api 통신 로직을 간결하게 하고 디버깅하기 위해서 설치하였습니다.

smb0123 committed:
feat: tanstack query 기본 설정

tanstack query를 사용하기 위한 기본 설정을 하였습니다.

smb0123 committed:
feat: tanstack query 기본 설정

tanstack query를 사용하기 위한 기본 설정을 하였습니다.

smb0123 committed:
feat: https 함수 추가

api와 관련된 공통 로직을 처리하는 https 함수를 추가하였습니다.
request interceptors에는 로컬 스토리지에 저장되어 있는 토큰을
headers의 Authorization에 담아서 보내는 로직이 있습니다.

response interceptors에는 응답의 상태코드가 401일 때 토큰을 갱신하는
로직이 있습니다.

smb0123 committed:
feat: https 함수 추가

api와 관련된 공통 로직을 처리하는 https 함수를 추가하였습니다.
request interceptors에는 로컬 스토리지에 저장되어 있는 토큰을
headers의 Authorization에 담아서 보내는 로직이 있습니다.

response interceptors에는 응답의 상태코드가 401일 때 토큰을 갱신하는
로직이 있습니다.

smb0123 committed:
feat: 카카오 로그인 api 로직 추가

useKakaoLogin, useKakaoLoginMutate 훅을 함수형으로 작성해보았습니다.

smb0123 committed:
feat: 카카오 로그인 api 로직 추가

useKakaoLogin, useKakaoLoginMutate 훅을 함수형으로 작성해보았습니다.

smb0123 committed:
chore: 폴더 구조 변경

login 페이지에서 사용하는 Section 컴포넌트를 login 폴더에 넣었습니다.

smb0123 committed:
chore: 폴더 구조 변경

login 페이지에서 사용하는 Section 컴포넌트를 login 폴더에 넣었습니다.

smb0123 committed:
chore: class-variance-authority 라이브러리 설치

VariantProps 유틸리티를 사용해서 컴포넌트의 props를 자동 추론, 객체
구조로 css 관리, 가독성 향상을 위해서 사용하였습니다.

smb0123 committed:
chore: class-variance-authority 라이브러리 설치

VariantProps 유틸리티를 사용해서 컴포넌트의 props를 자동 추론, 객체
구조로 css 관리, 가독성 향상을 위해서 사용하였습니다.

smb0123 committed:
feat: 소셜 로그인 로딩 페이지 구현

소셜 로그인 시 리다이렉트되는 페이지를 구현하였습니다.

smb0123 committed:
feat: 소셜 로그인 로딩 페이지 구현

소셜 로그인 시 리다이렉트되는 페이지를 구현하였습니다.

smb0123 committed:
feat: useSaveToken 훅 구현

쿼리 파라미터로 accessToken, refreshToken을 로컬 스토리지에 저장 후 로컬
스토리지에 토큰이 저장된 게 확인되면 특정 페이지로 이동

smb0123 committed:
feat: useSaveToken 훅 구현

쿼리 파라미터로 accessToken, refreshToken을 로컬 스토리지에 저장 후 로컬
스토리지에 토큰이 저장된 게 확인되면 특정 페이지로 이동

smb0123 committed:
refactor: Flex 컴포넌트의 props 변경

className props를 제거하고 height props를 추가하였습니다.
props를 명확하게 하기 위해서 입니다.

smb0123 committed:
refactor: Flex 컴포넌트의 props 변경

className props를 제거하고 height props를 추가하였습니다.
props를 명확하게 하기 위해서 입니다.

smb0123 committed:
refactor: Flex 컴포넌트의 props 변경

className props를 제거하고 height props를 추가하였습니다.
props를 명확하게 하기 위해서 입니다.

smb0123 committed:
refactor: Flex 컴포넌트의 props 변경

className props를 제거하고 height props를 추가하였습니다.
props를 명확하게 하기 위해서 입니다.

geongyu09 commented on src/apis/https.ts:
```diff
@@ -0,0 +1,36 @@
+import axios from "axios";
+
+export const https = axios.create({
```
개인적으로 https 라는 이름이 추상적인 느낌을 주는 것 같습니다!
로컬에서는 이 요청이 http일 것이고, 또 다른 인스턴스를 만들면 그것은 어떤 이름을 지어야 하는지 애매해지는 것 같아요!
https보다는 다른 이름을 짓는 것에 대해서는 어떻게 생각하시나요?

geongyu09 replied:
다른 여러 인스턴스도 만들 것을 대비해서 /instances 디랙토리를 따로 만들어서 사용해도 좋을 것 같습니다
여기서는  instance마다 각각의 파일을 만들어서 코드를 작성하고, index 로 한 곳에 모아서 re-export 하는 방식으로 해도 좋을 것 같습니다!

smb0123 replied:
다른 이름을 짓는 것에 매우 동의합니다.
axiosInstance는 어떤가요?

smb0123 replied:
instances 라는 디렉토리를 만들고 index.ts 파일을 만들어서 해당 파일안에 여러 함수들을 export 하는 것과 비교해서 건규님께서 말씀해주신 방법은 어떠한 장점이 있나요? (파일명으로 인스턴스의 이름을 알 수 있다는 장점인가요?)

geongyu09 replied:
파일명으로 아는 것도 있겠지만, 
하나의 인스턴스를 정의할 때에도 인스턴스 정의 + 인터셉터들 (혹은 이에 더해서 추가적으로 사용될 수 있는 함수들) 이 한 파일에 있게 되는데, 이게 여러 인스턴스가 한 파일에 있게 되면 그만큼 파일이 너무나도 복잡해질 것 같아요

아마 reissue 기능을 추가하게 된다면, 이는 무한루프를 피하기 위해서 다른 인스턴스를 사용해야만 할거에요
이렇게 두 개를 index 파일 하나에 만들어도 인스턴스 2개, 인터셉터 4개, 특정 인터셉터에 들어갈 함수들 n개 ... 가 될텐데, 너무 복잡도가 높아질 것이 우려됩니다.!

smb0123 replied:
오 그렇겠네요. 알겠습니다!
>아마 reissue 기능을 추가하게 된다면, 이는 무한루프를 피하기 위해서 다른 인스턴스를 사용해야만 할거에요

이 부분은 제 코드에서 볼 수 있듯이 재요청하는 api에 _retry 속성의 값을 true로 설정하여서 조건문을 통해 무한 루프를 피할 수 있습니다. (인스턴스를 하나만 사용해도 됩니다.)
```
if (error.response?.status === 401 && !originalRequest._retry) {
      originalRequest._retry = true;

      // Todo: 토큰 갱신 로직 추가
    }
```

smb0123 replied:
다른 방법으로는 건규님께서 말씀해주신 방법대로 다른 인스턴스를 사용하는 방법도 있겠네요!

geongyu09 commented on src/components/common/entities/Providers/index.tsx:
```diff
@@ -0,0 +1,40 @@
+  return new QueryClient({
+    defaultOptions: {
+      queries: {
+        staleTime: 60 * 1000,
```
staletime의 적용 근거가 궁금합니다!

smb0123 replied:
되게 날카로운 질문이네요. 보고 깜짝 놀랐습니다.
staletime에 대해서 생각을 해보진 않았고 tanstack query 공식 홈페이지에서 제공하는 초기 세팅 예시를 바탕으로 코드를 작성하였습니다. 이에 대해 생각을 해보지 못하였는데 다시 생각해 보겠습니다.
https://tanstack.com/query/latest/docs/framework/react/guides/advanced-ssr

smb0123 replied:
일단 1분으로 설정해보고 프로젝트를 진행해보면서 불편한 점이 있다면 수정해나가는 방향으로 가는 게 좋을 것 같습니다.

geongyu09 commented on src/components/common/entities/Providers/index.tsx:
```diff
@@ -0,0 +1,40 @@
+  }
+}
+
+export default function Providers({ children }: { children: React.ReactNode }) {
```
개인적으로 providers 보다는 더욱 명확한 이름을 사용하면 좋을 것 같습니다! 
가령 예시로 다른 provider를 사용하게 된다면, 그 provider는 어떤 이름을 사용하야할지에 대해서 애매해지는 문제점이 있는 것 같습니다!

geongyu09 commented on src/components/common/entities/Providers/index.tsx:
```diff
@@ -0,0 +1,40 @@
+      <ReactQueryDevtools initialIsOpen={false} />
+    </QueryClientProvider>
+  );
+}
```
공식문서를 잘 찾아보신 느낌이 많이 들었습니다..!! 진짜 대박 👍👍👍👍👍👍👍

geongyu09 commented on src/components/common/shared/layout/Flex/index.tsx:
```diff
@@ -7,6 +7,7 @@ interface FlexProps {
   alignItems?: AlignItemsType;
   maxWidth?: string;
   gap?: number;
+  height?: number | "h-full" | "h-screen" | "h-min" | "h-max" | "h-fit";
```
tailwind의 고질적인 문제점이 이렇게 타입을 하나하나 직접 입력해줘야한다는 것인 것 같아요.
좀더 멋진 방법이 있는지 많은 고민이 필요할 것 같습니다

widgh, height 값 모두 지속적으로 반복될 수 있을 것 같은데, 따로 빼는것도 좋을 것 같긴 합니다!
(다른 tailwind 관련 속성들 다..)
어떻게 생각하시나요..?

geongyu09 replied:
따로 빼지 않더라도, 코드 일관성을 위해서 height는 타입으로 정의 후 사용하는 것이 좋을 것 같습니다!

smb0123 replied:
되게 좋은 방법 같습니다!

geongyu09 commented on src/components/features/pages/social-login-loading/SpinnerSection/index.tsx:
```diff
@@ -0,0 +1,22 @@
+import Spinner from "@/components/common/shared/ui/Spinner";
+import { useSaveToken } from "@/hooks/social-login-loading";
+
+export default function SpinnerSection() {
```
혹시 폴더명을 케밥 케이스로 한 이유가 있나요?? 
기존에 컴포넌트 이름과 동일하게 하는 것으로 했던 것 같은데, 케밥 케이스를 사용했을 때 얻을 수 있는 장점이 있을까요?

smb0123 replied:
social-login-loading => 이 부분을 말씀하신 거라면 제가 이해하기로는 pages폴더 하위에는 각 페이지 폴더로 구분하는 것으로 알고 있어서 해당 페이지 명을 그대로 입력하였습니다. (페이지명을 케밥 케이스로 작성하는 것을 컨벤션으로 정하였기 때문에 이와 동일하게 가져가는 것이 가독성에 좋다고 생각했습니다.)

geongyu09 replied:
아..! 제가 착각했나보네요! 헐 이럴수가 🤦‍♂️

geongyu09 commented on src/hooks/login/index.ts:
```diff
@@ -0,0 +1,34 @@
+
+export const useKakaoLoginMutate = () => {
+  const { mutate } = useMutation({
+    mutationKey: ["kakaoLogin"],
```
혹시 mutationKey가 여기서는 어떻게 사용되는지 궁금합니다!

geongyu09 replied:
추가적으로, key를 관리할 수 있는 방법에 대해서 많은 고민이 필요할 수 있을 것 같습니다
개인적으로는 요청 uri와 동일하게 가도 괜찮지 않을까 싶습니다! 

이렇게 되면 네트워크를 보지 않아도 요청에 대한 응답을 devTool로 확인이 가능합니다!
물론 최대한 순수하게 query문을 사용해야만 합니다. 

다른 분들은 어떻게 생각하시나요?

smb0123 replied:
디버깅을 위해서 추가를 하였습니다!
사실 네트워크 탭으로 확인을 해볼 수도 있는 것이라서 다른 분들의 의견에 따르겠습니다.
저는 개인적으로 이번 프로젝트에서 devtool을 잘 활용해 보고 싶어서 mutationKey를 사용하고 싶습니다.
![image](https://github.com/user-attachments/assets/09ce4d6e-2b81-4252-867a-8be1565e7a34)

smb0123 replied:
key를 요청 uri와 동일하게 하는 것에 찬성합니다!
혹시 순수하게 query문을 사용해야 한다는 것이 이해가 잘 가지 않아서 간단하게 설명해주실 수 있을까요?

geongyu09 replied:
오 대박! 좋네요

순수하게 query문을 작성한다는 것은, 말 그대로 api문을 순수하게 요청한다는 뜻입니다. 
단순히 query요청을 하는게 아니라 이에 더불어서 다른 기능이나, 받아온 값을 다르게 캐싱을 한다거나 하는 경우라면, 네트워크창과 devtool의 값이 달라서 오히려 혼란을 주거나 문제점을 파악하는데 더 많은 시간을 소모할 것 같습니다!
딱 query문은 query문만을 담당하도록 하는거죠..!
어찌보면 api문을 순수하게 작성하는게 더 맞다고 볼 수도 있겠네요!

만약 그러지 못한다면, key에 어떤 처리가 들어갔는지를 구분할 수 있는 무언가를 추가적으로 줘야 할 것 같습니다

geongyu09 commented on src/hooks/login/index.ts:
```diff
@@ -0,0 +1,34 @@
+    ) => {
+      mutate(undefined, {
+        ...mutateOption,
+      });
```
이 방법 꽤 괜찮네요! 복잡하게 mutate의 콜백을 받지 않아도 되는 장점이 있는 것 같습니다!

하지만 혹시 mutate에 주는 인자가 정해져있지 않는 경우는 어떻게 사용해야하나요?

smb0123 replied:
mutate에 주는 인자라는게 혹시 MutateOptions에 대한 이야기인가요?

geongyu09 replied:
지금은 아무 인자가 없어서 undefined 를 주었지만, post요청에 인자를 주어야 하는 경우(mutateFn가 인자를 받는 경우)에는 어떻게 사용할 수 있을까요?

smb0123 replied:
postKakaoLogin 함수에 매개변수를 추가하고 useKakaoLogin 훅이 반환하는 onKakaoLogin에서 인자를 받아야 합니다.
이렇게 하면 depth가 한 단계 더 생기는군요..!

geongyu09 commented on src/hooks/login/index.ts:
```diff
@@ -0,0 +1,34 @@
+import { postKakaoLogin } from "@/apis/login";
+import { MutateOptions, useMutation } from "@tanstack/react-query";
+
+export const useKakaoLogin = () => {
```
컨밴션에 관련한 이야긴데요, 개인적인 의견으로는 한 파일에는 하나의 훅만 존재하도록 하는 것이 좋을 것 같습니다! 
로그인과 관련한 모든 훅을 login에 넣기에는 많은 훅(요청과 핸들러가 있는 훅, 요청이 있는 훅)이 한 파일에 전부 다 있는 것 보다는 각각을 책임에 맞추어서 분리하는 것에 대해서는 어떻게 생각하시나요?

smb0123 replied:
전 좋을 것 같습니다.
그렇다면 index.ts로 전부 imoprt 한 후 re-export 하는 방식으로 하는 게 좋을 것 같아요.

geongyu09 commented on src/hooks/social-login-loading/index.ts:
```diff
@@ -0,0 +1,29 @@
+        localStorageAccessToken === accessToken &&
+        localStorageRefreshToken === refreshToken
+      ) {
+        router.push("/");
```
path와 관련한 값은 하드코딩 보다는 상수 데이터를 사용하는 것이 더 좋을 것 같습니다! 
상수 데이터는 /constants 에 넣어두는 것이 좋을 것 같습니다!

smb0123 replied:
넵 알겠습니다!

geongyu09 commented on src/apis/login/index.ts:
```diff
@@ -0,0 +1,7 @@
+// Todo: 카카오 로그인 API 경로 변경
+
+export const postKakaoLogin = async () => {
+  await https.post("auth/users/signup", {});
```
uri는 하드코딩보다는 상수 데이터로 따로 관리하는 것이 좋을 것 같습니다!

추가적으로 axios를 사용하는 방법은 크게 2가지로 나눈다면,
- 위처럼 .post() 메서드를 사용하는 방식 
- axios에 method를 포함한 모든 데이터를 객체로 제공하는 방식

이 있다고 생각합니다!
axios 공식문서에서도 조금 더 메서드를 사용하는 전자의 방식을 추천하므로, axios 사용에 대한 컨밴션을 메서드를 사용하는 방식으로 정하는건 어떻게 생각하시나요?

smb0123 replied:
그렇다면 uri와 query key를 하나의 상수로 관리하는 게 좋겠네요!
저도 메서드를 사용하는 방식이 더 좋습니다!

geongyu09 reviewed (changes requested):
시험기간에도 수고 많으셨습니다!
컴포넌트 추상화가 정말로 마음에 드네요!

수정 사항(도 있지만,)보다는 다른 팀원들의 의견을 물어보는 질문들을 몇가지 남겨봤어요!
여러분들의 의견이 궁금합니다!

smb0123 committed:
chore: 폴더 구조 변경

https.ts 파일을 instances 폴더를 생성해서 안에 넣고 index.ts 파일을
생성해서 한곳에 모아서 re-export 하는 방식으로 변경하였습니다.

smb0123 committed:
chore: 폴더 구조 변경

https.ts 파일을 instances 폴더를 생성해서 안에 넣고 index.ts 파일을
생성해서 한곳에 모아서 re-export 하는 방식으로 변경하였습니다.

smb0123 committed:
rename: Providers 컴포넌트명 변경

의미가 명확해지도록 TanstackQueryProvider로 이름을 변경하였습니다.

smb0123 committed:
rename: Providers 컴포넌트명 변경

의미가 명확해지도록 TanstackQueryProvider로 이름을 변경하였습니다.

smb0123 committed:
feat: ROUTE 상수 추가

경로를 상수로 추가하여 하드 코딩을 제거하였습니다.

smb0123 committed:
feat: ROUTE 상수 추가

경로를 상수로 추가하여 하드 코딩을 제거하였습니다.

smb0123 committed:
chore: 폴더 구조 변경

훅을 하나의 파일에서 작성하고 index.ts 파일에서 re-export 하는 방법으로
변경하였습니다.

smb0123 committed:
chore: 폴더 구조 변경

훅을 하나의 파일에서 작성하고 index.ts 파일에서 re-export 하는 방법으로
변경하였습니다.

smb0123 committed:
fix: 빌드 에러 수정

useSearchParams() 훅은 Suspense 컴포넌트안에서 호출되어야 하기 때문에
수정하였습니다.

smb0123 committed:
fix: 빌드 에러 수정

useSearchParams() 훅은 Suspense 컴포넌트안에서 호출되어야 하기 때문에
수정하였습니다.

seunghyeon77 committed:
remove: package-lock.json 파일 삭제

seunghyeon77 committed:
refactor: InputProps 타입 변경, children 삭제

seunghyeon77 committed:
refactor: ButtonProps 타입 변경

seunghyeon77 committed:
refactor: className, children props 삭제

smb0123 committed:
rename: https 이름 변경

axiosInstance라는 더 명확한 이름으로 변경하였습니다.

smb0123 committed:
rename: https 이름 변경

axiosInstance라는 더 명확한 이름으로 변경하였습니다.

smb0123 committed:
chore: 폴더 구조 변경

훅을 한 파일당 한 개씩만 작성하고 훅들을 index.ts 파일에서 import 해서
re-export하는 방법으로 변경하였습니다.

smb0123 committed:
chore: 폴더 구조 변경

훅을 한 파일당 한 개씩만 작성하고 훅들을 index.ts 파일에서 import 해서
re-export하는 방법으로 변경하였습니다.

smb0123 committed:
refactor: useKakaoLoginMutate 훅에서 useMutatino 자체를 리턴하도록 변경

불필요한 코드를 제거할 수 있습니다.

smb0123 committed:
refactor: useKakaoLoginMutate 훅에서 useMutatino 자체를 리턴하도록 변경

불필요한 코드를 제거할 수 있습니다.

seunghyeon77 committed:
chore: pnpm-lock.yaml 파일 생성

seunghyeon77 committed:
design: 버튼 ui 컴포넌트 패딩 수정

seunghyeon77 committed:
refactor: 중복되는 Input을 TextField ui 컴포넌트로 변경

seunghyeon77 committed:
refactor: InputProps에서 type, placeholder props 삭제

seunghyeon77 committed:
refactor: 중복되는 Input을 TextField 컴포넌트로 변경

seunghyeon77 committed:
design: 페이지 레이아웃 여백 위치 변경

seunghyeon77 committed:
feat: Button UI 컴포넌트에 fullWidth props 추가

seunghyeon77 committed:
refactor: 페이지 전체 여백을 따로 상단 div에서 관리

seunghyeon77 committed:
refactor: TextField 컴포넌트에서 children 대신 방향 props에 JSX 넘겨주기

seunghyeon77 committed:
design: 버튼이 fullWidth 시 패딩 p-4로 변경

seunghyeon77 committed:
chore: EmailAuthSection 요소로 변경

seunghyeon77 committed:
chore: 컴포넌트 경로 변경

import 컴포넌트들의 경로를 상대 경로에서 절대 경로로 변경합니다.

#3

seunghyeon77 committed:
chore: 절대 경로 alias 변경

파일 절대 경로 alias를 @shared, @pages, @entities 등으로 변경합니다.

#3

seunghyeon77 committed:
chore: section요소 추가

EmailAuthSection에 section 요소 추가합니다.

#3

smb0123 committed:
chore: 변수명 변경

axiosInstance에서 instance로 변수명을 변경하였습니다.

smb0123 committed:
chore: 변수명 변경

axiosInstance에서 instance로 변수명을 변경하였습니다.

smb0123 committed:
chore: baseURL에 / 제거

NEXT_PUBLIC_BASE_URL에 /를 추가하고 baseURL에서 /를 제거하였습니다.

smb0123 committed:
chore: baseURL에 / 제거

NEXT_PUBLIC_BASE_URL에 /를 추가하고 baseURL에서 /를 제거하였습니다.

smb0123 committed:
refactor: 요청 api uri와 mutationKey를 상수 데이터로 관리

smb0123 committed:
refactor: 요청 api uri와 mutationKey를 상수 데이터로 관리

smb0123 committed:
refactor: useKakaoLoginMutate 훅을 query 폴더로 분리

smb0123 committed:
refactor: useKakaoLoginMutate 훅을 query 폴더로 분리

smb0123 committed:
Merge branch 'feat/6-login-api' of https://github.com/JNU-econovation/New-Project-FE into feat/6-login-api
